.github/workflows/php.yaml

```diff
@@ -82,9 +82,108 @@ jobs:
+          xvfb-run --server-args="-screen 0, 1280x720x24" --auto-servernum java -jar build/selenium-server-standalone.jar -log logs/selenium-server.log &
+
+      - name: Start ChromeDriver
+        if: ${{ !matrix.selenium-server && matrix.browser == 'chrome' }}
+        run: |
+          google-chrome --version
+          xvfb-run --server-args="-screen 0, 1280x720x24" --auto-servernum \
+          chromedriver --port=4444 --url-base=/wd/hub &> ./logs/chromedriver.log &
+
+      - name: Start GeckoDriver
+        if: ${{ !matrix.selenium-server && matrix.browser == 'firefox' }}
+        run: |
+          firefox --version
+          geckodriver --version
+          xvfb-run --server-args="-screen 0, 1280x720x24" --auto-servernum \
+          geckodriver &> ./logs/geckodriver.log &
+
+      - name: Start local PHP server
+        run: |
+          php -S 127.0.0.1:8000 -t tests/functional/web/ &>>./logs/php-server.log &
+
+      - name: Wait for browser & PHP to start
+        run: |
+          while ! nc -z localhost 4444 </dev/null; do echo Waiting for remote driver to start...; sleep 1; done
+          while ! nc -z localhost 8000 </dev/null; do echo Waiting for PHP server to start...; sleep 1; done
+
+      - name: Run tests
+        env:
+          BROWSER_NAME: ${{ matrix.browser }}
+          # When running directly against Firefox, we test only using geckodriver (not against legacy Firefox =<45), so we must declare GECKODRIVER=1
+          GECKODRIVER: "${{ (matrix.browser == 'firefox' && !matrix.selenium-server) && '1' || '0' }}"
+          # We must provide CHROMEDRIVER_PATH so that ChromeDriverTest is able to start Chromedriver itself
+          CHROMEDRIVER_PATH: "${{ (matrix.browser == 'chrome' && !matrix.selenium-server) && '/usr/local/share/chrome_driver/chromedriver' || '' }}"
+          DISABLE_W3C_PROTOCOL: "${{ matrix.w3c && '0' || '1' }}"
+        run: |
+          if [ "$BROWSER_NAME" = "chrome" ]; then EXCLUDE_GROUP+="exclude-chrome,"; fi
+          if [ "$BROWSER_NAME" = "firefox" ]; then EXCLUDE_GROUP+="exclude-firefox,"; fi
+          if [ -n "$EXCLUDE_GROUP" ]; then EXTRA_PARAMS+=" --exclude-group $EXCLUDE_GROUP"; fi
+          vendor/bin/phpunit --testsuite functional --colors=always --coverage-clover ./logs/clover.xml $EXTRA_PARAMS
+
+      - name: Submit coverage to Coveralls
+        # We use php-coveralls library for this, as the official Coveralls GitHub Action lacks support for clover reports:
+        # https://github.com/coverallsapp/github-action/issues/15
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_PARALLEL: true
+          COVERALLS_FLAG_NAME: ${{ github.job }}-${{ matrix.browser }}-selenium-${{ matrix.selenium-server }}-w3c-${{ matrix.w3c }}
+        run: |
+          composer global require php-coveralls/php-coveralls
+          ~/.composer/vendor/bin/php-coveralls -v
+
+      - name: Print logs
+        if: ${{ always() }}
```

Comment:
I used success or failure because it seemed like cancel would run this step

Reply:
But if the build is cancelled by github because other build failed, you may still want this build to dump its logs, no?

Reply:
It would make sense, that said I was thinking that if you implement `[ci skip]` like I did for phpMyAdmin it could run and make no sense